nova/cmd/manage.py

```diff
@@ -812,6 +812,75 @@ def version(self):
         """Print the current database version."""
         print(migration.db_version(database='api'))
 
+    @args('--dry-run', action='store_true', dest='dry_run', default=False,
+          help='Show number of candidate request specs')
+    @args('--older-than', dest='older_than', default=90,
+          help='Days from today to begin purging')
+    @args('--max-number', dest='max_number', default=0,
+          help='Maximum number of request spec rows to consider')
+    @args('--until-complete', action='store_true', dest='until_complete',
+          default=False,
+          help=('Run continuously until all rows are purged. Use --max-number '
+                'as a batch size for each iteration.'))
+    def purge_request_specs(self, dry_run=False, older_than=90, max_number=0,
+                            until_complete=False):
+        """Removes old request specs for deleted instances"""
+        older_than = int(older_than)
+        max_number = int(max_number)
+
+        admin_context = context.get_admin_context()
+        extant_instance_uuids = [
+            # There's no need to join the default security_groups and
+            # instance_info_caches tables, hence explicit empty
+            # 'columns_to_join'.
+            i.get('uuid', None) for i in db.instance_get_all(
+                admin_context, columns_to_join=[])
+        ]
+        finished = False
+        while not finished:
+            finished = self._purge_request_specs_and_instance_mappings(
+                admin_context, extant_instance_uuids, dry_run, older_than,
```

Comment:
Shouldn't we filter out the `None`s out of `extant_instance_uuids` before passing on the list?

Reply:
i couldn't readily find any instance rows that didn't have a uuid, but i fixed it anyway.